src/main/kotlin/tech/relaycorp/poweb/PoWebClient.kt

```diff
@@ -212,7 +214,11 @@ public class PoWebClient internal constructor(
                     exc
                 )
             }
-            collectAndAckParcels(this, this@flow, trustedCertificates)
+            try {
+                collectAndAckParcels(this, this@channelFlow, trustedCertificates)
+            } catch (e: CancellationException) {
+                close(CloseReason(CloseReason.Codes.NORMAL, ""))
```

Comment:
I think this may be the reason why the test `Cancelling the flow should close the connection normally` is failing: ktor may have already started the closure. Can you please run that test with this change:

```suggestion
                return@wsConnect
```

On the other hand, if that does not fix the broken test, we should still  `return` after calling `close()` because we should ignore the server's close code in that case.

Reply:
Didn't fix the issue, so I've just added the return.